c++/test/TestColumnReader.cc

```diff
@@ -434,6 +434,89 @@ TEST(TestColumnReader, testDictionaryWithNulls) {
   }
 }
 
+TEST(TestColumnReader, testEncodedVectorWithNulls) {
```

Comment:
Add test to cover case when dictionary encoding is not used, but lazy decoding option is set. Also cover composite types, such as list, map, etc.